File: easy217.cpp
//
// Created by hello on 8/14/22.
//

#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;
bool containsDuplicate(vector<int>& nums) {
    unordered_map<int,int>mp;
    for (int i = 0; i < nums.size(); ++i) {
        if(mp.count(nums[i]))return true;
        mp[nums[i]]++;
    }
    return false;
}
int main(){
    vector<int>v1={1,1,3,3,4,3,2,4,2};
    bool ans=containsDuplicate(v1);
    cout<<ans<<endl;
}

File: easy217.java
package solutions.hash;

import java.util.HashMap;

public class easy217 {
    public static void main(String[] args) {
        boolean ans=containsDuplicate(new int[]{1,2,3,1});
        System.out.println(ans);
    }

    public static boolean containsDuplicate(int[] nums) {
        HashMap<Integer,Integer>map=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(map.containsKey(nums[i]))return true;
            map.put(nums[i],1);
        }
        return false;
    }
}
